src/runtime-rs/crates/hypervisor/src/dragonball/vmm_instance.rs

```diff
@@ -103,6 +104,10 @@ impl VmmInstance {
         result
     }
 
+    pub fn set_seccomp(&mut self, seccomp: HashMap<ThreadType, BpfProgram>) {
```

Comment:
I think hashmap is too heavy for this case

```rust
pub enum ScopedSeccomp {
    all: Option<BpfProgram>,
    vcpu: ...
    vmm: ...
}

pub struct VmmInstance {
    seccomp: ScopedSeccomp
}
```

Reply:
Reasons:
1. the types are not dynamic
2. in the foreseeable future we don't have too much seccomp scopes.